### PR TITLE
[BugFix] fix uri to support file path contains blank space & more fit hadoop uri

### LINKED ISSUE #8060: [Bug]Query external table with blank in hdfs file path
### Steps to reproduce the behavior (Required)
1. CREATE TABLE `table_one`
(  
 `dept_no` int,  
 `addr` string,  
 `tel` string
)
partitioned by(dt string) 
ROW FORMAT DELIMITED FIELDS TERMINATED BY ',' ; 
2. insert overwrite table table_one partition(dt='first dt') values ('new data','new user id','new scene');
3. SELECT * from table_one;

### Expected behavior (Required)
图片

### Real behavior (Required)
图片

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by feihengye**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8060

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaClient.java|1|1|100.00%|[]|

**Review by imay — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java [hunk 1]
```diff
@@ -674,7 +674,7 @@ public TextFileFormatDesc getTextFileFormatDesc(StorageDescriptor sd) {
 
     public List<HdfsFileDesc> getHdfsFileDescs(String dirPath, boolean isSplittable,
                                                StorageDescriptor sd) throws Exception {
-        URI uri = new URI(dirPath.replace(" ", "%20"));
+        URI uri = new Path(dirPath).toUri();
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java:677]: It's better to supplement the title.

Comment (feihengye) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java:677]: ok, please recheck

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by chaoyli** [comment]
@Mergifyio backport branch-2.2 branch-2.3 branch-2.4

**Comment by chaoyli** [comment]
run starrocks_admit_test

**Pull request merged**